applications/DEMApplication/python_scripts/materials_assignation_utility.py

```diff
@@ -0,0 +1,70 @@
+import KratosMultiphysics as Kratos
+from KratosMultiphysics.DEMApplication import *
+
+class MaterialsAssignationUtility:
+
+    def __init__(self, model, spheres_model_part, DEM_material_parameters):
+        self.model = model
+        self.spheres_model_part = spheres_model_part
+        self.DEM_material_parameters = DEM_material_parameters
+        self.read_materials_utility = Kratos.ReadMaterialsUtility(model)
+
+    def AssignMaterialParametersToProperties(self):
+
+        materials_parameters = self.DEM_material_parameters
+        list_of_materials = materials_parameters["materials"]
+        list_of_material_relations = materials_parameters["material_relations"]
+
+        for material in list_of_materials:
+            material_id = material["material_id"].GetInt()
+            if  not self.spheres_model_part.HasProperties(material_id):
+                self.spheres_model_part.AddProperties(Kratos.Properties(material_id))
+
+            properties_of_model_part_with_this_id = self.spheres_model_part.GetProperties()[material_id]
+            properties = material["Variables"]
+            properties_of_model_part_with_this_id[Kratos.PARTICLE_MATERIAL] = material_id
+            self.read_materials_utility.AssignMaterialToProperty(material, properties_of_model_part_with_this_id)
+
+            for material_relation in list_of_material_relations:
+                subprops = None
+                material_ids_list = material_relation["material_ids_list"].GetVector()
+                if material_id == material_ids_list[0]:
+                    index_of_the_other_material = int(material_ids_list[1])
+                    subprops = Kratos.Properties(index_of_the_other_material)
+                elif material_id == material_ids_list[1]:
+                    index_of_the_other_material = int(material_ids_list[0])
+                    subprops = Kratos.Properties(index_of_the_other_material)
+
+                if subprops:
+                    contact_properties = material_relation["Variables"]
+                    self.read_materials_utility.AssignMaterialToProperty(material_relation, subprops)
+
+                    discontinuum_constitutive_law_instance = globals().get(subprops[DEM_DISCONTINUUM_CONSTITUTIVE_LAW_NAME])()
```

Comment:
why don't you use [kratos globals](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_interface/kratos_globals.py#L138-L154) here?

=> we used this in the Core utility before it was moved to C++

Reply:
The plan here is to use **factories**, as everyone is doing currently. But I left it for a subsequent PR, as this one is very big already. That will allow us to avoid the 'import *' in this file. I have a reminder issue for this.